### PR TITLE
Update/expand comments

### PR DESCRIPTION
**Pull request by fsoikin**
Addresses #4

### CODE REVIEW

File: test/Main.purs [hunk 1]
```diff
@@ -61,7 +60,7 @@ spec = do
       testComponent def do
         find ".baz" >> do
           prop "value" >>= shouldEqual ""
-          simulate' "change" $ unsafeToForeign { target: { value: "New text" } }
+          simulate' "change" { target: { value: "New text" } }
```
Comment (fsoikin) [inline test/Main.purs:63]: According to [Enzyme docs](https://enzymejs.github.io/enzyme/docs/api/ReactWrapper/simulate.html), this parameter is not just any old value that gets passed straight to the handler. It's a record that gets merged into the synthetic event. So I changed it from `a` to `Record r`.

And this is also (I'm guessing) the reason that `simulate` only works for "standard" elements, not for custom events.

File: src/Elmish/Enzyme.purs [hunk 2]
```diff
@@ -144,45 +207,52 @@ is selector = E.is selector =<< ask
...
-setState :: forall state. state -> EnzymeM Unit
-setState newState =
-  E.setState newState =<< ask
+-- |
+-- | NOTE: this is a type-unsafe operation. There is no check to make sure the
+-- | state being set here has the type of the actual state the component in
+-- | question is using.
+unsafeSetState :: forall state. state -> EnzymeM Unit
+unsafeSetState newState =
+  E.unsafeSetState newState =<< ask
```
Comment (mcordova47) [inline src/Elmish/Enzyme.purs:219]: 👍🏻

File: src/Elmish/Enzyme.purs [hunk 3]
```diff
@@ -211,38 +281,33 @@ count selector = E.count selector =<< ask
 -- | re-render. For some reason, Enzyme won't pick up the changes automatically.
 -- | See https://enzymejs.github.io/enzyme/docs/api/ReactWrapper/update.html for
 -- | more info.
+-- |
+-- | NOTE: this only works on the "root" element, which means it cannot be
+-- | called inside `withSelector` or `withElement`.
 update :: EnzymeM Unit
-update =
-  E.update =<< ask
+update = E.update =<< ask
```
Comment (mcordova47) [inline src/Elmish/Enzyme.purs:288]: 👍🏻  We could probably enforce this in the types, too. Maybe `EnzymeM` could have an extra parameter indicating whether it’s the root.

File: test/Main.purs (hunk 1 above)
Comment (mcordova47) [reply test/Main.purs:63]: Oh didn’t realize, good catch

**Review by mcordova47 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**